src/components/OSCALControl.js

```diff
@@ -61,6 +61,27 @@ export default function OSCALControl(props) {
     );
   }
 
+  // Retrieve the set-parameters when contained in the top-level implemented requirement statement
+  let topLevelImplReqStatement = null;
+  if (props.implReqStatements) {
+    // NOTE: The top level statement ends with "_smt"
+    props.implReqStatements.forEach((statement, index) => {
+      if (statement["statement-id"].endsWith("_smt")) {
+        topLevelImplReqStatement =
+          props.implReqStatements[Object.keys(props.implReqStatements)[index]];
+      }
+    });
+  }
+  // NOTE: The top-level by-component is the first element in the by-components array
```

Comment:
I'm not sure this is an accurate statement.

Each `component` in the `by-components` array represents how each component address the control.

I believe `OSCALContorlImplementationImplReq` sends in a `componentId` that we should check for?

Reply:
I was a little hesitant to include that part as well. Grabbing the first array is definitely not the best practice here as this doesn't seem to be a consistent pattern. Also, thanks for clarifying on this. Your right about `OSCALControlImplementationImplReq` passing `componentId` as a prop. I can make a `by-component` id match check as well - I'm not seeing it elsewhere.